In [203]:
import pandas as pd

In [204]:
data=pd.read_csv('movies_metadata.csv')

C:\Users\SREEJA\AppData\Local\Temp\ipykernel_4008\3199850993.py:1: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  data=pd.read_csv('movies_metadata.csv')


In [205]:
data.shape

(45466, 24)

In [206]:
data.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

##### Factors that determine the success of a movie

In [207]:
data.adult

0        False
1        False
2        False
3        False
4        False
         ...  
45461    False
45462    False
45463    False
45464    False
45465    False
Name: adult, Length: 45466, dtype: object

- 1st column adult 

In [208]:
data.adult.unique()

array(['False', 'True', ' - Written by Ørnås',
       ' Rune Balot goes to a casino connected to the October corporation to try to wrap up her case once and for all.',
       ' Avalanche Sharks tells the story of a bikini contest that turns into a horrifying affair when it is hit by a shark avalanche.'],
      dtype=object)

 we can see 
 . false, true are in string format
 . 3 rows other than values False,True

In [209]:
# findout what rows are other than False,True
data.loc[~data.adult.isin(["False","True"])]

adult  \
19730                                 - Written by Ørnås   
29503   Rune Balot goes to a casino connected to the ...   
35587   Avalanche Sharks tells the story of a bikini ...   

      belongs_to_collection                            budget  \
19730              0.065736  /ff9qCepilowshEtG2GYWwzt2bs4.jpg   
29503              1.931659  /zV8bHuSL6WXoD6FWogP9j4x80bL.jpg   
35587              2.185485  /zaSf5OG7V8X8gqFvly88zDdRm46.jpg   

                                                  genres  \
19730  [{'name': 'Carousel Productions', 'id': 11176}...   
29503  [{'name': 'Aniplex', 'id': 2883}, {'name': 'Go...   
35587  [{'name': 'Odyssey Media', 'id': 17161}, {'nam...   

                                                homepage          id imdb_id  \
19730  [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...  1997-08-20       0   
29503  [{'iso_3166_1': 'US', 'name': 'United States o...  2012-09-29       0   
35587           [{'iso_3166_1': 'CA', 'name': 'Canada'}]  2014-01-01       0   

      original_language                            original_title  overview  \
19730             104.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   
29503              68.0      [{'iso_639_1': 'ja', 'name': '日本語'}]  Released   
35587              82.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   

       ... release_date revenue runtime spoken_languages status  tagline  \
19730  ...            1     NaN     NaN              NaN    NaN      NaN   
29503  ...           12     NaN     NaN              NaN    NaN      NaN   
35587  ...           22     NaN     NaN              NaN    NaN      NaN   

       title video vote_average vote_count  
19730    NaN   NaN          NaN        NaN  
29503    NaN   NaN          NaN        NaN  
35587    NaN   NaN          NaN        NaN  

[3 rows x 24 columns]

In [210]:
# checking for other columns we can see 
# belongs_to_collection cannot be numeric, budget can be numeric, id cannot be data,language cannot be numeric
# DECIDED TO DROP THE ROWS

In [211]:
data=data.loc[data.adult.isin(["False","True"]),:]

In [212]:
data.adult.unique()

array(['False', 'True'], dtype=object)

In [213]:
# false, true are in string format 
# need to convert them into boolean

In [214]:
data.adult=data.adult.map({'True':True,'False':False})

In [215]:
data.adult.unique()

array([False,  True])

In [216]:
data.shape

(45463, 24)

### 2nd column belongs_to_collection

In [217]:
data.belongs_to_collection

0        {'id': 10194, 'name': 'Toy Story Collection', ...
1                                                      NaN
2        {'id': 119050, 'name': 'Grumpy Old Men Collect...
3                                                      NaN
4        {'id': 96871, 'name': 'Father of the Bride Col...
                               ...                        
45461                                                  NaN
45462                                                  NaN
45463                                                  NaN
45464                                                  NaN
45465                                                  NaN
Name: belongs_to_collection, Length: 45463, dtype: object

In [218]:
data.belongs_to_collection[2]

"{'id': 119050, 'name': 'Grumpy Old Men Collection', 'poster_path': '/nLvUdqgPgm3F85NMCii9gVFUcet.jpg', 'backdrop_path': '/hypTnLot2z8wpFS7qwsQHW1uV8u.jpg'}"

In [219]:
data.belongs_to_collection.isnull().sum()

40972

 - data type of belongs_to_collection is string format dictionary
 problems  need tobe solved - 
 1. convert the string to dictionary (its original form)
 2. extract 'name' from the dictionary
 3. replace NaN value with No Collection (since we have 40972)
 4. replace entire belongs column with collection column

In [220]:
import ast
collections=[]
for i in data.belongs_to_collection:
    if i==i:
        collections.append(ast.literal_eval(i)['name'])
    else:
        collections.append("No collection")
        

In [221]:
collections

['Toy Story Collection',
 'No collection',
 'Grumpy Old Men Collection',
 'No collection',
 'Father of the Bride Collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'James Bond Collection',
 'No collection',
 'No collection',
 'Balto Collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'Ace Ventura Collection',
 'No collection',
 'Chili Palmer Collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'Babe Collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'No collection',
 'Mortal Kombat Collection',
 'No collection',
 'No collection',
 'No collection',
 'Pocahontas Collection',
 'No collection',
 'No collection',
 'No collecti

In [222]:
data['Collections']=collections

In [223]:
data=data.drop(['belongs_to_collection'],axis=1)

In [224]:
data.head()

adult    budget                                             genres  \
0  False  30000000  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  False  65000000  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  False         0  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  False  16000000  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4  False         0                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview popularity  ...  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943  ...   
1  When siblings Judy and Peter discover an encha...  17.015539  ...   
2  A family wedding reignites the ancient feud be...    11.7129  ...   
3  Cheated on, mistreated and stepped on, the wom...   3.859495  ...   
4  Just when George Banks has recovered from his ...   8.387519  ...   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  \
0                    Toy Story  False          7.7     5415.0   
1                      Jumanji  False          6.9     2413.0   
2             Grumpier Old Men  False          6.5       92.0   
3            Waiting to Exhale  False          6.1       34.0   
4  Father of the Bride Part II  False          5.7      173.0   

                      Collections  
0            Toy Story Collection  
1                   No collection  
2       Grumpy Old Men Collection  
3                   No collection  
4  Father of the Bride Collection  

[5 rows x 24 columns]

#### Budget

In [225]:
data.budget.value_counts()[0]

36573

In [226]:
# %of zero values
sum(data.budget=='0')*100/data.shape[0]

80.44563711149726

1. there are 36573 null values in budget column
2. format of budget is not correct
3. 80 percent of budget is not having null values

##### i am not taking budget as a factor for movie success

##### Genre

In [227]:
data.genres[0]

"[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]"

In [228]:
data.genres.isnull().sum()

0

1. genre is in string dictionary format
2. each movie belongs to multiple genres


##### HomePage

In [229]:
data.homepage.isnull().sum()

37684

In [230]:
data.homepage.isnull()

0        False
1         True
2         True
3         True
4         True
         ...  
45461    False
45462     True
45463     True
45464     True
45465     True
Name: homepage, Length: 45463, dtype: bool

In [231]:
data['HomePagePresent']=(~data.homepage.isnull())

In [232]:
data=data.drop(['homepage'],axis=1)

In [233]:
data.head()

adult    budget                                             genres     id  \
0  False  30000000  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...    862   
1  False  65000000  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   8844   
2  False         0  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  15602   
3  False  16000000  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  31357   
4  False         0                     [{'id': 35, 'name': 'Comedy'}]  11862   

     imdb_id original_language               original_title  \
0  tt0114709                en                    Toy Story   
1  tt0113497                en                      Jumanji   
2  tt0113228                en             Grumpier Old Men   
3  tt0114885                en            Waiting to Exhale   
4  tt0113041                en  Father of the Bride Part II   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943   
1  When siblings Judy and Peter discover an encha...  17.015539   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...   3.859495   
4  Just when George Banks has recovered from his ...   8.387519   

                        poster_path  ... runtime  \
0  /rhIRbceoE9lR4veEXuwCC2wARtG.jpg  ...    81.0   
1  /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg  ...   104.0   
2  /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg  ...   101.0   
3  /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg  ...   127.0   
4  /e64sOI48hQXyru7naBFyssKFxVd.jpg  ...   106.0   

                                    spoken_languages    status  \
0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
1  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   
2           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
3           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
4           [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                             tagline  \
0                                                NaN   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  \
0                    Toy Story  False          7.7     5415.0   
1                      Jumanji  False          6.9     2413.0   
2             Grumpier Old Men  False          6.5       92.0   
3            Waiting to Exhale  False          6.1       34.0   
4  Father of the Bride Part II  False          5.7      173.0   

                      Collections HomePagePresent  
0            Toy Story Collection            True  
1                   No collection           False  
2       Grumpy Old Men Collection           False  
3                   No collection           False  
4  Father of the Bride Collection           False  

[5 rows x 24 columns]

##### ID

In [234]:
data.shape[0]-len(data.id.unique())

30

In [235]:
data=data.drop_duplicates(subset="id")

In [236]:
data.id.unique()

array(['862', '8844', '15602', ..., '67758', '227506', '461257'],
      dtype=object)

In [237]:
data=data.dropna(subset="id")

In [238]:
data.id

0           862
1          8844
2         15602
3         31357
4         11862
          ...  
45461    439050
45462    111109
45463     67758
45464    227506
45465    461257
Name: id, Length: 45433, dtype: object

In [239]:
data.id.dtypes

dtype('O')

In [240]:
# converting to integers
data.id=data.id.astype('int64')

In [241]:
data.id.dtypes

dtype('int64')

##### imbd_id

In [242]:
data.imdb_id[0]

'tt0114709'

In [243]:
data.imdb_id.isnull().sum()

17

1. every movie must have imdb_id
2. it cannot have no imdb_id

In [244]:
# drop duplicates of imdb
data=data.drop_duplicates(subset="imdb_id")

In [245]:
data.shape[0]-len(data.imdb_id.unique())

0

##### original language

In [246]:
data.original_language.unique()

array(['en', 'fr', 'zh', 'it', 'fa', 'nl', 'de', 'cn', 'ar', 'es', 'ru',
       'sv', 'ja', 'ko', 'sr', 'bn', 'he', 'pt', 'wo', 'ro', 'hu', 'cy',
       'vi', 'cs', 'da', 'no', 'nb', 'pl', 'el', 'sh', 'xx', 'mk', 'bo',
       'ca', 'fi', 'th', 'sk', 'bs', 'hi', 'tr', 'is', 'ps', 'ab', 'eo',
       'ka', 'mn', 'bm', 'zu', 'uk', 'af', 'la', 'et', 'ku', 'fy', 'lv',
       'ta', 'sl', 'tl', 'ur', 'rw', 'id', 'bg', 'mr', 'lt', 'kk', 'ms',
       'sq', nan, 'qu', 'te', 'am', 'jv', 'tg', 'ml', 'hr', 'lo', 'ay',
       'kn', 'eu', 'ne', 'pa', 'ky', 'gl', 'uz', 'sm', 'mt', 'hy', 'iu',
       'lb', 'si'], dtype=object)

In [247]:
len(data.original_language.unique())

90

In [248]:
len(data.original_language)

45417

In [249]:
data.original_language.dtypes

dtype('O')

In [250]:
data.original_language.isnull().sum()

11

In [251]:
data.loc[data.original_language.isnull(),:]

adult budget                                             genres  \
19574  False      0                [{'id': 99, 'name': 'Documentary'}]   
21602  False      0  [{'id': 10749, 'name': 'Romance'}, {'id': 18, ...   
22832  False      0  [{'id': 10752, 'name': 'War'}, {'id': 37, 'nam...   
32141  False      0                     [{'id': 35, 'name': 'Comedy'}]   
37407  False      0                  [{'id': 16, 'name': 'Animation'}]   
41047  False      0                [{'id': 99, 'name': 'Documentary'}]   
41872  False      0                [{'id': 99, 'name': 'Documentary'}]   
44057  False      0                      [{'id': 18, 'name': 'Drama'}]   
44410  False      0  [{'id': 16, 'name': 'Animation'}, {'id': 99, '...   
44576  False      0                                                 []   
44655  False      0  [{'id': 28, 'name': 'Action'}, {'id': 10752, '...   

           id    imdb_id original_language              original_title  \
19574  283101  tt0429086               NaN     Shadowing the Third Man   
21602  103902  tt0838231               NaN              Unfinished Sky   
22832  359195  tt0053558               NaN             13 Fighting Men   
32141  147050  tt0122580               NaN                   Lambchops   
37407  257095  tt0225145               NaN                      Bajaja   
41047  332742  tt4432912               NaN              Song of Lahore   
41872  144410  tt0154152               NaN  Annabelle Serpentine Dance   
44057  380438  tt0298411               NaN       Lettre d'une inconnue   
44410  381096  tt5333518               NaN                        Garn   
44576  381525  tt5376720               NaN                      WiNWiN   
44655  104473  tt0223958               NaN       La prise de Tournavos   

                                                overview popularity  \
19574  Documentary about the production of The Third ...   0.017007   
21602  An Outback farmer takes in an Afghani woman wh...   0.359818   
22832  A group of Union Army soldiers is charged with...   0.070647   
32141  George and Gracie enter an elegant drawing roo...   0.073418   
37407  The first fairy tale transformed into a full-l...   0.036841   
41047  Until the late 1970s, the Pakistani city of La...   0.373688   
41872  Two dancers perform in succession facing a sta...    1.58434   
44057                                                NaN   0.001283   
44410  The traditional crafts of crochet and knitting...   0.067624   
44576  American investment fund buys Austrian compani...   0.030766   
44655  Three military men, seen inside a fortificatio...   0.471325   

                            poster_path  ... runtime  \
19574  /xT9csxMKYbizRQgvNzCXJ67R7er.jpg  ...    95.0   
21602  /vfaJuZF0ocVwBEDYZhoULGuVxrz.jpg  ...    94.0   
22832  /q5ADXvvzYtAOyrReIg1HckdPj62.jpg  ...    69.0   
32141  /upQByEwY58sHWdL3RZm7kmRc44s.jpg  ...     8.0   
37407  /12wf2EkrPGxNzRbNyoau8GIFp3M.jpg  ...    87.0   
41047  /6iJqscs2xQLKKvrTCWmlHd9rf3W.jpg  ...    82.0   
41872  /l7Lzvah6PL6vdMF5p0elKiyYqp9.jpg  ...     1.0   
44057  /gk32ZWTYgfTXctEyCDAWeMcxo7X.jpg  ...     0.0   
44410  /olEOCEGTW15WBPC0UnmLjokrmNF.jpg  ...    76.0   
44576  /pErCRQoc3nt8lCsExd3fFYgyBox.jpg  ...    84.0   
44655  /4VmQBE9qBRxCE3gHtSAa5AUsURq.jpg  ...     1.0   

                                        spoken_languages    status  tagline  \
19574  [{'iso_639_1': 'de', 'name': 'Deutsch'}, {'iso...  Released      NaN   
21602           [{'iso_639_1': 'en', 'name': 'English'}]  Released      NaN   
22832           [{'iso_639_1': 'en', 'name': 'English'}]  Released      NaN   
32141           [{'iso_639_1': 'en', 'name': 'English'}]  Released      NaN   
37407             [{'iso_639_1': 'cs', 'name': 'Český'}]  Released      NaN   
41047  [{'iso_639_1': 'ur', 'name': 'اردو'}, {'iso_63...  Released      NaN   
41872       [{'iso_639_1': 'xx', 'name': 'No Language'}]  Released      NaN   
44057          [{'iso_639_1': 'fr', 'name': 'Français'}]  Released     

In [252]:
data.original_language=data.original_language.fillna('mode')

In [253]:

data.loc[data.original_language.isnull(),:]


Empty DataFrame
Columns: [adult, budget, genres, id, imdb_id, original_language, original_title, overview, popularity, poster_path, production_companies, production_countries, release_date, revenue, runtime, spoken_languages, status, tagline, title, video, vote_average, vote_count, Collections, HomePagePresent]
Index: []

[0 rows x 24 columns]

##### original_title

In [254]:
data.shape[0]-len(data.original_title.unique())

2060

In [255]:
(data.original_title.isnull().sum())

0

In [256]:
len(data.original_title.unique())

43357

In [257]:
data.original_title=data.original_title.str.strip()

In [258]:
data.original_title=data.original_title.str.lower()

In [259]:
data.original_title

0                          toy story
1                            jumanji
2                   grumpier old men
3                  waiting to exhale
4        father of the bride part ii
                    ...             
45461                        رگ خواب
45462            siglo ng pagluluwal
45463                       betrayal
45464            satana likuyushchiy
45465                       queerama
Name: original_title, Length: 45417, dtype: object

##### overview

In [260]:
data.shape[0]-len(data.overview.unique())

1124

In [261]:
data.overview.isnull().sum()

952

In [262]:
data.loc[data.overview.isnull(),:]

adult    budget                                             genres  \
32     False         0  [{'id': 10749, 'name': 'Romance'}, {'id': 12, ...   
300    False  22000000  [{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...   
634    False         0                     [{'id': 35, 'name': 'Comedy'}]   
635    False         0                     [{'id': 35, 'name': 'Comedy'}]   
641    False         0                     [{'id': 35, 'name': 'Comedy'}]   
...      ...       ...                                                ...   
45342  False         0                      [{'id': 18, 'name': 'Drama'}]   
45377  False         0                  [{'id': 12, 'name': 'Adventure'}]   
45398  False   1254040  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
45399  False    750000  [{'id': 80, 'name': 'Crime'}, {'id': 35, 'name...   
45410  False         0  [{'id': 18, 'name': 'Drama'}, {'id': 80, 'name...   

           id    imdb_id original_language                    original_title  \
32      78802  tt0114952                fr  guillaumet, les ailes du courage   
300    161495  tt0114296                sv                         roommates   
634    287305  tt0117312                de    peanuts – die bank zahlt alles   
635    339428  tt0116485                de                     happy weekend   
641     10801  tt0117788                de                     das superweib   
...       ...        ...               ...                               ...   
45342  199887  tt1771636                en                        over/under   
45377  317389  tt0070695                es     simbad e il califfo di bagdad   
45398  468707  tt5742932                fi  lauri mäntyvaaran tuuheet ripset   
45399  280422  tt3805180                ru                       все и сразу   
45410  449131  tt0321264                ru                             aprel   

      overview popularity                       poster_path  ... runtime  \
32         NaN   0.745542  /k6ODtR38dKEfuzSGjggr8KDyAF4.jpg  ...    50.0   
300        NaN   3.395867  /hvHNlMvWS2GBt7RR971bJ3k4bJc.jpg  ...   108.0   
634        NaN   0.066123  /wpk30SvRHmjC2plgKHZXxG0FlKd.jpg  ...     NaN   
635        NaN   0.002229                               NaN  ...     NaN   
641        NaN   0.821299  /AbhMKCh3fV5PY2B9uSPF1DWEvq2.jpg  ...    86.0   
...        ...        ...                               ...  ...     ...   
45342      NaN   0.704642  /1xLaIBGGPE4APtBJdfeuyOWICZ0.jpg  ...    87.0   
45377      NaN   0.006352  /izk7KbT6LZO9baEhCkOZYMgj60w.jpg  ...     NaN   
45398      NaN   0.347806  /rKOpJuwb7pTqYVShHM2tl25VxyF.jpg  ...    90.0   
45399      NaN   0.201582  /hNsmPpl3zLG36jr4EIEd5P8I4pa.jpg  ...     0.0   
45410      NaN   0.008903                               NaN  ...     NaN   

                                spoken_languages    status  tagline  \
32      [{'iso_639_1': 'en', 'name': 'English'}]  Released      NaN   
300     [{'iso_639_1': 'en', 'name': 'English'}]  Released      NaN   
634                                           []  Released      NaN   
635     [{'iso_639_1': 'de', 'name': 'Deutsch'}]  Released      NaN   
641     [{'iso_639_1': 'de', 'name': 'Deutsch'}]  Released      NaN   
...                                          ...       ...      ...   
45342   [{'iso_639_1': 'en', 'name': 'English'}]  Released      NaN   
45377  [{'iso_639_1': 'it', 'name': 'Italiano'}]  Released      NaN   
45398     [{'iso_639_1': 'fi', 'name': 'suomi'}]  Released      NaN   
45399   [{'iso_639_1': 'ru', 'name': 'Pусский'}]  Released      NaN   
45410                                         []  Released      NaN   

                                  title  video vote_average vote_count  \
32                     Wings of Courage  False          6.8        4.0   
300                           Roommates  False          6.4        7.0   
634      Peanuts – Die Bank zahlt alles  False          4.0        1.0   
635                       Happy Weekend  False          0.0

In [263]:
data.overview=data.overview.fillna('no overview available')

In [264]:
data.loc[data.overview.isnull(),:]

Empty DataFrame
Columns: [adult, budget, genres, id, imdb_id, original_language, original_title, overview, popularity, poster_path, production_companies, production_countries, release_date, revenue, runtime, spoken_languages, status, tagline, title, video, vote_average, vote_count, Collections, HomePagePresent]
Index: []

[0 rows x 24 columns]

##### popularity

In [265]:
data.popularity.isnull().sum()

3

In [266]:
data.popularity.dtypes

dtype('O')

In [267]:
data.popularity=data.popularity.astype('float64')

In [268]:
data.loc[data.popularity.isnull(),:]

adult budget                                             genres  \
19729  False      0  [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   
29502  False      0  [{'id': 16, 'name': 'Animation'}, {'id': 878, ...   
35586  False      0  [{'id': 10770, 'name': 'TV Movie'}, {'id': 28,...   

           id    imdb_id original_language    original_title  \
19729   82663  tt0113002                en      midnight man   
29502  122662  tt2423504                ja  マルドゥック・スクランブル 排気   
35586  249260  tt2622826                en  avalanche sharks   

                                                overview  popularity  \
19729  British soldiers force a recently captured IRA...         NaN   
29502         Third film of the Mardock Scramble series.         NaN   
35586  A group of skiers are terrorized during spring...         NaN   

      poster_path  ... runtime spoken_languages status  tagline  title video  \
19729         NaN  ...     NaN              NaN    NaN      NaN    NaN   NaN   
29502         NaN  ...     NaN              NaN    NaN      NaN    NaN   NaN   
35586         NaN  ...     NaN              NaN    NaN      NaN    NaN   NaN   

      vote_average vote_count                  Collections HomePagePresent  
19729          NaN        NaN                No collection           False  
29502          NaN        NaN  Mardock Scramble Collection            True  
35586          NaN        NaN                No collection           False  

[3 rows x 24 columns]

1. popularity null values have no vote_average, vote_count

In [269]:
data=data.dropna(subset="popularity")

In [270]:
data.loc[data.popularity.isnull(),:]

Empty DataFrame
Columns: [adult, budget, genres, id, imdb_id, original_language, original_title, overview, popularity, poster_path, production_companies, production_countries, release_date, revenue, runtime, spoken_languages, status, tagline, title, video, vote_average, vote_count, Collections, HomePagePresent]
Index: []

[0 rows x 24 columns]

##### poster path

In [271]:
# poster path does not make difference in success of movie

In [272]:
data=data.drop(["poster_path"],axis=1)

##### run time

In [283]:
data.runtime[0:10]

0     81.0
1    104.0
2    101.0
3    127.0
4    106.0
5    170.0
6    127.0
7     97.0
8    106.0
9    130.0
Name: runtime, dtype: float64

In [284]:
data.runtime.isnull().sum()

255

In [285]:
data.loc[data.runtime.isnull(),:]

adult budget                                             genres  \
634    False      0                     [{'id': 35, 'name': 'Comedy'}]   
635    False      0                     [{'id': 35, 'name': 'Comedy'}]   
644    False      0                      [{'id': 18, 'name': 'Drama'}]   
802    False      0                      [{'id': 18, 'name': 'Drama'}]   
863    False      0                   [{'id': 53, 'name': 'Thriller'}]   
...      ...    ...                                                ...   
45246  False      0                     [{'id': 35, 'name': 'Comedy'}]   
45310  False      0                                                 []   
45313  False      0                      [{'id': 18, 'name': 'Drama'}]   
45377  False      0                  [{'id': 12, 'name': 'Adventure'}]   
45410  False      0  [{'id': 18, 'name': 'Drama'}, {'id': 80, 'name...   

           id    imdb_id original_language                  original_title  \
634    287305  tt0117312                de  peanuts – die bank zahlt alles   
635    339428  tt0116485                de                   happy weekend   
644    278978  tt0118026                de       und keiner weint mir nach   
802    282919  tt0112865                de                       diebinnen   
863    253632  tt0094822                fr                     baton rouge   
...       ...        ...               ...                             ...   
45246  231216  tt0441908                de                 villa henriette   
45310  418757  tt4112020                pl         między nami dobrze jest   
45313  369444  tt0098035                pl                 ostatni dzwonek   
45377  317389  tt0070695                es   simbad e il califfo di bagdad   
45410  449131  tt0321264                ru                           aprel   

                    overview  popularity  \
634    no overview available    0.066123   
635    no overview available    0.002229   
644    no overview available    0.439989   
802    no overview available    0.106345   
863    no overview available    0.437895   
...                      ...         ...   
45246  no overview available    0.002513   
45310  no overview available    0.030803   
45313  no overview available    0.000102   
45377  no overview available    0.006352   
45410  no overview available    0.008903   

                                    production_companies  ... runtime  \
634    [{'name': 'Westdeutscher Rundfunk (WDR)', 'id'...  ...     NaN   
635     [{'name': 'Senator Film Produktion', 'id': 191}]  ...     NaN   
644                                                   []  ...     NaN   
802                                                   []  ...     NaN   
863                                                   []  ...     NaN   
...                                                  ...  ...     ...   
45246                                                 []  ...     NaN   
45310                                                 []  ...     NaN   
45313                                                 []  ...     NaN   
45377  [{'name': 'Roas Produzioni', 'id': 21137}, {'n...  ...     NaN   
45410                                                 []  ...     NaN   

                                spoken_languages    status  tagline  \
634                                           []  Released      NaN   
635     [{'iso_639_1': 'de', 'name': 'Deutsch'}]  Released      NaN   
644     [{'iso_639_1': 'de', 'name': 'Deutsch'}]  Released      NaN   
802                                           []  Released      NaN   
863                                           []  Released      NaN   
...                                          ...       ...      ...   
45246   [{'iso_639_1': 'de', 'name': 'Deutsch'}]  Released      NaN   
45310                                         []  Released      NaN   
45313                                         []  Released      NaN   
45377  [{'iso_639_1': 'it', 'name': 'Italiano'}]  Released      N

In [288]:
data.loc[data.runtime==0,'runtime']=90

In [289]:
data.runtime.isnull().sum()

255

##### spoken_languages

In [294]:
data.spoken_languages[99]

"[{'iso_639_1': 'es', 'name': 'Español'}, {'iso_639_1': 'en', 'name': 'English'}]"

In [296]:
Total_Languages=[]
for i in data.spoken_languages.values:
    prod=ast.literal_eval(i)
    Total_Languages.append(len(prod))

In [298]:
data['Total_languages']=Total_Languages

In [299]:
data=data.drop(["spoken_languages"],axis=1)

In [300]:
data

adult    budget                                             genres  \
0      False  30000000  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1      False  65000000  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2      False         0  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3      False  16000000  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4      False         0                     [{'id': 35, 'name': 'Comedy'}]   
...      ...       ...                                                ...   
45461  False         0  [{'id': 18, 'name': 'Drama'}, {'id': 10751, 'n...   
45462  False         0                      [{'id': 18, 'name': 'Drama'}]   
45463  False         0  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   
45464  False         0                                                 []   
45465  False         0                                                 []   

           id    imdb_id original_language               original_title  \
0         862  tt0114709                en                    toy story   
1        8844  tt0113497                en                      jumanji   
2       15602  tt0113228                en             grumpier old men   
3       31357  tt0114885                en            waiting to exhale   
4       11862  tt0113041                en  father of the bride part ii   
...       ...        ...               ...                          ...   
45461  439050  tt6209470                fa                      رگ خواب   
45462  111109  tt2028550                tl          siglo ng pagluluwal   
45463   67758  tt0303758                en                     betrayal   
45464  227506  tt0008536                en          satana likuyushchiy   
45465  461257  tt6980792                en                     queerama   

                                                overview  popularity  \
0      Led by Woody, Andy's toys live happily in his ...   21.946943   
1      When siblings Judy and Peter discover an encha...   17.015539   
2      A family wedding reignites the ancient feud be...   11.712900   
3      Cheated on, mistreated and stepped on, the wom...    3.859495   
4      Just when George Banks has recovered from his ...    8.387519   
...                                                  ...         ...   
45461        Rising and falling between a man and woman.    0.072051   
45462  An artist struggles to finish his work while a...    0.178241   
45463  When one of her hits goes wrong, a professiona...    0.903007   
45464  In a small town live two brothers, one a minis...    0.003503   
45465  50 years after decriminalisation of homosexual...    0.163015   

                                    production_companies  ... runtime  \
0         [{'name': 'Pixar Animation Studios', 'id': 3}]  ...    81.0   
1      [{'name': 'TriStar Pictures', 'id': 559}, {'na...  ...   104.0   
2      [{'name': 'Warner Bros.', 'id': 6194}, {'name'...  ...   101.0   
3      [{'name': 'Twentieth Century Fox Film Corporat...  ...   127.0   
4      [{'name': 'Sandollar Productions', 'id': 5842}...  ...   106.0   
...                                                  ...  ...     ...   
45461                                                 []  ...    90.0   
45462             [{'name': 'Sine Olivia', 'id': 19653}]  ...   360.0   
45463  [{'name': 'American World Pictures', 'id': 6165}]  ...    90.0   
45464               [{'name': 'Yermoliev', 'id': 88753}]  ...    87.0   
45465                                                 []  ...    75.0   

         status                                            tagline  \
0      Released                                                NaN   
1      Released          Roll the dice and unleash the excitement!   
2      Released  Still Yelling. Still Fighting. Still Ready for...   
3      Released  Friends are the people who let you be yourself...   
4      Released  Just When His World Is Back To Normal... He's ...   
...         ...    

##### status

In [301]:
data.status.value_counts()

Released           44971
Rumored              227
Post Production       98
In Production         20
Planned               15
Canceled               2
Name: status, dtype: int64

In [302]:
data.status.isnull().sum()

81

In [303]:
data.loc[data.status.isnull(),:]

adult budget                                             genres  \
189    False      0                [{'id': 99, 'name': 'Documentary'}]   
767    False      0                [{'id': 99, 'name': 'Documentary'}]   
1384   False      0                     [{'id': 35, 'name': 'Comedy'}]   
2140   False      0                                                 []   
3460   False      0                      [{'id': 18, 'name': 'Drama'}]   
...      ...    ...                                                ...   
44377  False      0                                                 []   
45067  False      0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
45090  False      0                                                 []   
45159  False      0                                                 []   
45442  False      0                                                 []   

           id    imdb_id original_language  \
189     56088  tt0114435                en   
767     52059  tt0113147                en   
1384    40506  tt0119644                en   
2140    77314  tt0101530                fr   
3460    94214  tt0210130                en   
...       ...        ...               ...   
44377   39827  tt0131064                en   
45067   49492  tt1653015                en   
45090  128268  tt0876565                en   
45159   41811  tt0099480                en   
45442   44330  tt0135690                en   

                                      original_title  \
189                                         the show   
767                       the gate of heavenly peace   
1384                               meet wally sparks   
2140                      the cabinet of dr. ramirez   
3460                      jails, hospitals & hip-hop   
...                                              ...   
44377           the who: thirty years of maximum r&b   
45067                          jogwa - the awakening   
45090  veruschka - die inszenierung (m)eines körpers   
45159                                         eating   
45442                           le tripot clandestin   

                                                overview  popularity  \
189    A documentary about the culture of hip-hop. Th...    0.159596   
767    The Gate of Heavenly Peace is a feature-length...    0.107050   
1384   Wally Sparks is a tabloid TV show reporter who...    0.576741   
2140                               no overview available    0.037500   
3460   Jails, Hospitals &amp; Hip-Hop is a cinematic ...    0.009057   
...                                                  ...         ...   
44377              A celebration of 30 years of The Who.    0.014260   
45067  Jogwa actually means alms given to a person, u...    0.004990   
45090         Documentary about Veruschka von Lehndorff.    0.056298   
45159  At a spacious house in Los Angeles, Helene is ...    0.000002   
45442  A combination gambling den and bawdy house is ...    0.219260   

      production_companies  ... runtime status  \
189                     []  ...    93.0    NaN   
767                     []  ...   180.0    NaN   
1384                    []  ...   105.0    NaN   
2140                    []  ...     NaN    NaN   
3460                    []  ...    90.0    NaN   
...                    ...  ...     ...    ...   
44377                   []  ...   155.0    NaN   
45067                   []  ...    92.0    NaN   
45090                   []  ...    52.0    NaN   
45159                   []  ...   110.0    NaN   
45442                   []  ...     3.0    NaN   

                                                 tagline  \
189        What Happens On Stage Is Only Half The Story!   
767                                                  NaN   
1384                                                 NaN   
2140                                                 NaN   
3460   three worlds / two million voices / one genera...   
...                                                  ...   
44377 

##### vote_average

In [274]:
data.vote_average.isnull().sum()

0

In [275]:
data.vote_average.dtypes

dtype('float64')

##### vote_count

In [277]:
data.vote_count.isnull().sum()

0

In [278]:
data.vote_count.dtypes


dtype('float64')

In [279]:
data.vote_count=data.vote_count.astype('int64')